### PR TITLE
Move the predictions into Celestial and Vessel

### PR DESCRIPTION
**Pull request by pleroy**
This is preparatory work for removing Ye Olde Ugly Boolean (YOUB).

### CODE REVIEW

File: ksp_plugin/celestial.hpp [hunk 1]
```diff
@@ -56,6 +61,12 @@ class Celestial {
   // Deletes the |prolongation_| and forks a new one at |time|.
   void ResetProlongation(Instant const& time);
 
+  // Creates a |prediction_| forked at the end of the |prolongation_|.
+  void ForkPrediction();
```
Comment (eggrobin) [inline ksp_plugin/celestial.hpp]: Document the whole contract

File: ksp_plugin/plugin.cpp [hunk 2]
```diff
@@ -286,7 +287,7 @@ Plugin::NewBodyCentredNonRotatingTransforms(
           reference_body_prolongation_or_prediction =
               [this, reference_body]() -> Trajectory<Barycentric> const& {
                   if (transforms_are_operating_on_predictions_) {
-                  return *FindOrDie(system_predictions_, reference_body);
+                  return reference_body->Celestial::prediction();
```
Comment (eggrobin) [inline ksp_plugin/plugin.cpp]: TODO, this is weird.
We don't really have a compiler at the moment though...

File: ksp_plugin/celestial.hpp [hunk 3]
```diff
@@ -46,6 +46,11 @@ class Celestial {
   Trajectory<Barycentric> const& prolongation() const;
   not_null<Trajectory<Barycentric>*> mutable_prolongation();
 
+  // Both accessors require |is_initialized()|.  In addition the first one
+  // requires that a |prediction_| currently exist.
+  Trajectory<Barycentric> const& prediction() const;
```
Comment (eggrobin) [inline ksp_plugin/celestial.hpp]: has_prediction, same in Vessel.

File: ksp_plugin/celestial.hpp (hunk 1 above)
Comment (pleroy) [inline ksp_plugin/celestial.hpp]: Done (2x).

File: ksp_plugin/plugin.cpp (hunk 2 above)
Comment (pleroy) [inline ksp_plugin/plugin.cpp]: Done.

File: ksp_plugin/celestial.hpp (hunk 3 above)
Comment (pleroy) [inline ksp_plugin/celestial.hpp]: Done.

**Pull request merged**